### DISCUSSION TITLE
WalletConnect and Metamask on web mobile - connection is not established

### DISCUSSION THREAD

**Discussion by onurio** [wevm/wagmi#1212] [Q&A]
Hi! Thanks for the great project!

I'm having trouble identifying what's the problem with my implementation.

1. On a mobile browser I open the wallet connect modal, and choose metamask. (see screenshot) iPhone 7 running iOS 15.7, also running latest wage and metamask app.

<img src="https://user-images.githubusercontent.com/36936789/199136178-36b172b2-564a-4baf-a4a7-aa2b732d5142.jpeg" width="250"/>

2. Metamask app opens, asks me to authenticate, but nothing happens after that. Any idea what could be the cause of this issue?

Also if I try to press the metamask button again it gives me this error (see screenshot)
<img src="https://user-images.githubusercontent.com/36936789/199136429-4129579c-fa22-4a62-a000-d241f6397248.jpeg" width="250"/>

Client implementation

```jsx
const { chains, provider, webSocketProvider } = configureChains(defaultChains, [
  publicProvider(),
]);

export const wagmiClient = createClient({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({
      chains,
      options: {
        shimDisconnect: true,
        UNSTABLE_shimOnConnectSelectAccount: true,
      },
    }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
    new InjectedConnector({
      chains,
      options: {
        name: 'Injected',
        shimDisconnect: true,
      },
    }),
  ],
  provider,
  webSocketProvider,
});

const WagmiProvider = ({ children }) => (
  <WagmiConfig client={wagmiClient}>{children}</WagmiConfig>
);
```

**On desktop everything works perfectly**

The example on your website for connecting a wallet works perfectly.. But I also copied the same code. So I'm not sure what I'm doing

**The only difference is that this app is rendered inside an iframe**

**Comment by kobyle69** [comment]
Same issue, did you ever fixed it?
